### ISSUE TITLE
Unicode font issues in terminal

### ISSUE THREAD

**Issue by franckrasolo** [CypherpunkArmory/UserLAnd#246]
Similarly to what has previously been reported against https://github.com/corbinlc/GNURootDebian/issues/93,

I cannot get patched Powerline fonts to display as expected in the terminal and associated editor.

Expected rendering (taken from [vim-airline](https://github.com/vim-airline/vim-airline)):
![vim-airline-demo](https://raw.githubusercontent.com/wiki/vim-airline/vim-airline/screenshots/demo.gif)

Actual rendering (on my Android device):
![screenshot](https://user-images.githubusercontent.com/94347/45244123-48e92680-b2ee-11e8-8cb2-2fea12671a61.jpg)

*Device Information*

Device: Samsung Galaxy Tab S3
Android Version: Oreo 8.0.0
UserLAnd Version: 0.3.4

**Comment by franckrasolo** [comment]
Hi @lapsio,

It seems unicode is indeed supported by the ConnectBot client used with UserLAnd.
It is already set to use `xterm-256color` and so is my `$TERM` environment variable as you can see below:

![env_term](https://user-images.githubusercontent.com/94347/45267294-ad78c280-b461-11e8-9758-9671c9481b5f.jpg)

I have also ensured that `LANG` and all `LC_*` environment variables are set to a `UTF-8` locale:

![locales](https://user-images.githubusercontent.com/94347/45267295-bf5a6580-b461-11e8-8c31-6d462cd24e52.jpg)

I installed `fontconfig` in my user profile and configured it to select exactly three specific fonts:

![fonts](https://user-images.githubusercontent.com/94347/45267219-53c3c880-b460-11e8-8e89-7e503aefa33e.jpg)

However I am unsure how to change the default font.

The above changes still did not yield the expected results. To further troubleshoot this, I printed the following to illustrate the issue:

![unicode_characters](https://user-images.githubusercontent.com/94347/45267254-f4b28380-b460-11e8-9378-a6a86e357eb1.jpg)

**Comment by franckrasolo** [comment]
Well, the only<sup>[[1]](#footnote1)</sup> solution to this problem actually turned out to be Termux with `openssh`.

With Termux, not only am I able to select any given font from internal/external storage, but I can also customise the terminal colours declaratively.

For posterity, it may help others to know how I fixed this:
1. install Termux
2. install the OTF/TTF font(s) you want to use:
```
mkdir -p ~/.termux
cp -v /sdcard/Download/<selected-font>.[ot]tf ~/.termux/
cp ~/.termux/<selected-font>.[ot]tf ~/.termux/font.ttf
termux-reload-settings
```
3. assuming the UserLAnd service is running:
```
pkg install openssh
ssh -p 2022 user@localhost
```

I now have a half-decent setup as illustrated below:

_vim-airline tabline_
![termux_vim-airline_tabline](https://user-images.githubusercontent.com/94347/45335326-b9a27400-b576-11e8-9a7e-08884535992d.jpg)

_vim-airline statusline_
![termux_vim-airline_statusline](https://user-images.githubusercontent.com/94347/45335302-9d063c00-b576-11e8-85dc-c17906a53597.jpg)

<a name="note1">[1]</a>: To date, none of Termius, JuiceSSH, Connectbot, VX Connectbot allowed the selection of fonts from device storage.

Over the past year, I transitioned from Termux to GNURoot Debian to UserLAnd, so I didn't expect to be returning to Termux to solve this one.
